core/utils/exact_match_table.h

```diff
@@ -309,6 +309,10 @@ class ExactMatchTable {
   // Returns the ith field.
   const ExactMatchField &get_field(size_t i) const { return fields_[i]; }
 
+  typename EmTable::iterator begin() { return table_.begin(); }
+
+  typename EmTable::iterator end() { return table_.end(); }
```

Comment:
I had to make these non-`const` iterators, and I'm not 100% sure why. Should I be using `cbegin` and `cend` here?

Reply:
You could use `cbegin` or `cend`, but then they would still technically be non-`const` as they would be `const_iterator`s, but either way is fine I think.

Reply:
(Perhaps we should add a `const_iterator` to cuckoo map?)

Reply:
I've looked into adding `const_iterator` to cuckoo map, and it seems excessively annoying (the two methods seem to be: write the iterator twice, once as non-const and once as const; or, write a template in place of the two versions of the iterator, then expand it twice to make the non-const and const iterators).